src/lib.rs

```diff
@@ -1139,41 +1144,49 @@ impl BlazeSymbolizer {
                     return vec![];
                 };
 
-                let sym = resolver.find_symbol(*addr);
+                let res_syms = resolver.find_symbols(*addr);
                 let linfo = if self.line_number_info {
                     resolver.find_line_info(*addr)
                 } else {
                     None
                 };
-                if sym.is_none() && linfo.is_none() {
-                    vec![]
-                } else if sym.is_none() {
-                    let linfo = linfo.unwrap();
-                    vec![SymbolizedResult {
-                        symbol: "".to_string(),
-                        start_address: 0,
-                        path: linfo.path,
-                        line_no: linfo.line_no,
-                        column: linfo.column,
-                    }]
-                } else if let Some(linfo) = linfo {
-                    let (sym, start) = sym.unwrap();
-                    vec![SymbolizedResult {
-                        symbol: String::from(sym),
-                        start_address: start,
-                        path: linfo.path,
-                        line_no: linfo.line_no,
-                        column: linfo.column,
-                    }]
+                if res_syms.is_empty() {
+                    if linfo.is_some() {
+                        let linfo = linfo.unwrap();
+                        vec![SymbolizedResult {
+                            symbol: "".to_string(),
+                            start_address: 0,
+                            path: linfo.path,
+                            line_no: linfo.line_no,
+                            column: linfo.column,
+                        }]
+                    } else {
+                        vec![]
+                    }
                 } else {
-                    let (sym, start) = sym.unwrap();
-                    vec![SymbolizedResult {
-                        symbol: String::from(sym),
-                        start_address: start,
-                        path: "".to_string(),
-                        line_no: 0,
-                        column: 0,
-                    }]
+                    let mut results = vec![];
+                    for sym in res_syms {
```

Comment:
Not really sure I understand the signature of this `symbolize` method. Why does it return nested vectors and not just a single one?

Reply:
It symbolizes a list of addresses.  So, it needs an entry for each address.  However, several symbols can be at the same address for reasons.  One of the reasons is optimization, LTO for example, can merge functions.